<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TxtEditor</class>
 <widget class="QWidget" name="TxtEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>522</width>
    <height>422</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QPlainTextEdit" name="plainTextEdit">
     <property name="acceptDrops">
      <bool>false</bool>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="plainText">
      <string notr="true"/>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::TextEditorInteraction</set>
     </property>
     <property name="backgroundVisible">
      <bool>false</bool>
     </property>
     <property name="centerOnScroll">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="tabPosition">
      <enum>QTabWidget::North</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <attribute name="title">
       <string>Navi</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_22">
         <property name="text">
          <string>&lt;</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_23">
         <property name="text">
          <string>Up</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="pushButton_24">
         <property name="text">
          <string>&gt;</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>PgUp</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QLineEdit" name="linenum_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="5">
        <widget class="QLabel" name="totalline_3">
         <property name="text">
          <string>/99999</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_25">
         <property name="text">
          <string>Left</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_26">
         <property name="text">
          <string>Down</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="pushButton_27">
         <property name="text">
          <string>Right</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="pushButton_2">
         <property name="text">
          <string>PgDn</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QPushButton" name="pushButton_38">
         <property name="text">
          <string>goto</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Tool</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_36">
         <property name="text">
          <string>Redo</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="save_button">
         <property name="text">
          <string>Save</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_4">
         <property name="text">
          <string>Paste</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="pushButton_37">
         <property name="text">
          <string>Quit</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="pushButton_5">
         <property name="text">
          <string>Cut</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_34">
         <property name="text">
          <string>Undo</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_3">
         <property name="text">
          <string>Copy</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Search</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3" columnstretch="0,1,0,0">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Search</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="search_lineedit"/>
         </item>
         <item>
          <widget class="QCheckBox" name="case_checkbox">
           <property name="text">
            <string>Case</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="pushButton_7">
         <property name="font">
          <font>
           <pointsize>6</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Prev</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QPushButton" name="pushButton_8">
         <property name="font">
          <font>
           <pointsize>6</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Next</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Replace</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="replace_lineedit"/>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="pushButton_9">
         <property name="font">
          <font>
           <pointsize>6</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Replace</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="pushButton_10">
         <property name="font">
          <font>
           <pointsize>6</pointsize>
          </font>
         </property>
         <property name="text">
          <string>ReplaceAll</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>plainTextEdit</sender>
   <signal>cursorPositionChanged()</signal>
   <receiver>TxtEditor</receiver>
   <slot>setlinenum()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>plainTextEdit</sender>
   <signal>blockCountChanged(int)</signal>
   <receiver>TxtEditor</receiver>
   <slot>totalline(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>255</x>
     <y>268</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>229</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_23</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>up()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>71</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_26</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>down()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>pageup()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>198</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>pagedown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_25</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>left()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>31</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_27</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>right()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_38</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>gotoline()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>177</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>findnext()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>459</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>79</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>findprev()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_9</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>replace()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>364</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_10</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>replaceall()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>459</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>93</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save_button</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>savefile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>379</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>56</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>plainTextEdit</sender>
   <signal>textChanged()</signal>
   <receiver>TxtEditor</receiver>
   <slot>textchanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>255</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_34</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>102</x>
     <y>58</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_36</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_37</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>501</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>108</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_22</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>prev()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_24</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>next()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>204</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>-5</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>-14</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>TxtEditor</receiver>
   <slot>cut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>319</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>578</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>pageup()</slot>
  <slot>pagedown()</slot>
  <slot>left()</slot>
  <slot>right()</slot>
  <slot>up()</slot>
  <slot>down()</slot>
  <slot>undo()</slot>
  <slot>redo()</slot>
  <slot>setlinenum()</slot>
  <slot>gotoline()</slot>
  <slot>totalline(int)</slot>
  <slot>findnext()</slot>
  <slot>findprev()</slot>
  <slot>replace()</slot>
  <slot>replaceall()</slot>
  <slot>savefile()</slot>
  <slot>textchanged()</slot>
  <slot>prev()</slot>
  <slot>next()</slot>
  <slot>copy()</slot>
  <slot>cut()</slot>
  <slot>paste()</slot>
 </slots>
</ui>
